webapp/src/ts/effects/contacts.effects.ts

```diff
@@ -93,6 +93,7 @@ export class ContactsEffects {
               this.globalActions.setSnackbarContent(this.translateService.instant('error.404.title'));
             }
             console.error('Error selecting contact', err);
+            this.contactsActions.updateSelectedContactSummary({ errorStack: err.stack });
```

Comment:
@latin-panda catching the error inside `loadContactSummary` meant that this catch block here wouldnt execute
so to kinda harmonize everything i decided to updatedSummary in this catch block. please let me know what you think about this. thanks

Reply:
Thanks for the update!
This function has many processes going on, and putting it here will retrieve all the errors of any of those processes, which is also not in the scope of this ticket. 

You can try this: 
- add a catch block in the loadContactSummary, 
- inside that block call updateSelectedContactSummary
- then throw the error again

Like this, it will keep the usage of `updateSelectedContactSummary` inside the same function; that's good because it keeps it contained.  And by throwing the error, it will execute the catch block in this `selectContact` effect. 

```javascript

private loadContactSummary(contactId, trackName) {
    const trackPerformance = this.performanceService.track();
    const selected = this.selectedContact;
    return this.contactSummaryService
      .get(selected.doc, selected.reports, selected.lineage, selected.targetDoc)
      .then(summary => {
        return this
          .verifySelectedContactNotChanged(contactId)
          .then(() => {
            this.contactsActions.setContactsLoadingSummary(false);
            return this.contactsActions.updateSelectedContactSummary(summary);
          });
      })
      .catch(error => {
         this.contactsActions.updateSelectedContactSummary({ errorStack: error.stack });
         throw error;
      })
      .finally(() => {
        trackPerformance?.stop({ name: [ ...trackName, 'load_contact_summary' ].join(':') });
      });
  }
```

Reply:
:) awesome